### PR TITLE
Option Added CURL_LOCK_DATA_HOSTS_NAME

### PR DESCRIPTION
**Pull request by mickae1**
Option Added:
CURL_LOCK_DATA_HOSTS_NAME:
When Transfering multiple file with SSH2, the access of the known_host file is not protected.
To protect the access of the file, use this option to protect the access of the file
Be careful, only one file is authorized. Because the handle of the file is shared across connections

I don't know how to make a pull request only on one Commit ! 

The commit  on Aug 06, 2020 is already pulled.

### CODE REVIEW

**Comment by mickae1** [comment]
I don't know how to make a pull request only on one Commit !

The commit on Aug 06, 2020 is already pulled.

**Comment by bagder** [comment]
you can run `git rebase -i` in your branch, and fixup/squash them as you think is fine, then force-push...

File: docs/libcurl/curl_share_setopt.3 [hunk 1]
```diff
@@ -94,6 +94,11 @@ The Public Suffix List stored in the share object is made available to all
 easy handle bound to the later. Since the Public Suffix List is periodically
 refreshed, this avoids updates in too many different contexts.
 
+.IP CURL_LOCK_DATA_HOSTS_NAME
+When Transfering multiple file with SSH2, the access of the known_host file is not protected. 
+To protect the access of the file, use this option to protect the access of the file
+Be careful, only one file is authorized. Because the handle of the file is shared across connections
+
```
Comment (bagder) [inline docs/libcurl/curl_share_setopt.3:101]: I'm not sure I understand. What do you mean with "only one file is authorized" ? 

Also, you should mention that this feature only works with the libssh2 backend unless you intend to fix the others as well.

File: docs/libcurl/curl_share_setopt.3 [hunk 2]
```diff
@@ -94,6 +94,11 @@ The Public Suffix List stored in the share object is made available to all
 easy handle bound to the later. Since the Public Suffix List is periodically
 refreshed, this avoids updates in too many different contexts.
 
+.IP CURL_LOCK_DATA_HOSTS_NAME
```
Comment (bagder) [inline docs/libcurl/curl_share_setopt.3:97]: Maybe call it `CURL_LOCK_DATA_KNOWNHOSTS` or did I misunderstand its purpose?

File: docs/libcurl/symbols-in-versions [hunk 3]
```diff
@@ -873,6 +873,7 @@ CURL_LOCK_DATA_NONE             7.10.3
 CURL_LOCK_DATA_PSL              7.61.0
 CURL_LOCK_DATA_SHARE            7.10.4
 CURL_LOCK_DATA_SSL_SESSION      7.10.3
+CURL_LOCK_DATA_HOSTS_NAME       7.72.1
```
Comment (bagder) [inline docs/libcurl/symbols-in-versions:925]: The next release (the earliest time this can be merged) is 7.74.0

File: lib/vssh/libssh2.c [hunk 4]
```diff
@@ -451,6 +452,9 @@ static CURLcode ssh_knownhost(struct connectdata *conn)
     int keycheck = LIBSSH2_KNOWNHOST_CHECK_FAILURE;
     int keybit = 0;
 
+    if(data->share && (data->share->specifier & (1<< CURL_LOCK_DATA_HOSTS_NAME)) )
+        Curl_share_lock(data, CURL_LOCK_DATA_HOSTS_NAME, CURL_LOCK_ACCESS_SINGLE);
```
Comment (bagder) [inline lib/vssh/libssh2.c:458]: wrong indent level

File: lib/vssh/libssh2.c [hunk 5]
```diff
@@ -605,6 +609,8 @@ static CURLcode ssh_knownhost(struct connectdata *conn)
       }
       break;
     }
+    if(data->share && (data->share->specifier & (1<< CURL_LOCK_DATA_HOSTS_NAME)) )
+        Curl_share_unlock(data, CURL_LOCK_DATA_HOSTS_NAME);
```
Comment (bagder) [inline lib/vssh/libssh2.c:615]: also wrong indent

**Pull request closed without merging**